src/centos/scripts/start-marklogic.sh

```diff
@@ -155,16 +157,47 @@ function restart_check {
 #   $3 :  Additional options to pass to curl
 ################################################################
 function curl_retry_validate {
-    for ((i = 0; i < N_RETRY; i = i + 1)); do
+    local return_response_code="${4:-false}"
+    local retry_count=0
+    while true; do
         request="curl -m 30 -s -w '%{http_code}' $3 $1"
         response_code=$(eval "${request}")
         if [[ ${response_code} -eq $2 ]]; then
-            return 0
+            break
+        else
+            retry_count=$((retry_count + 1))
+            if [[ $retry_count -ge $N_RETRY ]]; then
+                if [[ "${return_response_code}" = "true" ]]; then
+                    return "${response_code}"
+                else
+                    error "Expected response code ${2}, got ${response_code} from ${1}." exit
+                fi
+            fi
+            sleep ${RETRY_INTERVAL}
         fi
-        sleep ${RETRY_INTERVAL}
     done
 
-    error "Expected response code ${2}, got ${response_code} from ${1}." exit
+    return "${response_code}"
+}
+
+################################################################
+# Validates MarkLogic bootstrap host
+# input:  $1:        MarkLogic Bootstrap Host
+# returns valid:     if it's a valid MarkLogic bootstrap host
+#         invalid:   if it's not a valid MarkLogic bootstrap host
+#         localhost: if bootstrap host is the localhost
+################################################################
+function validate_bootstrap {
+    local bootsrap_host=$1
+    localhost_id=$(curl --anyauth -u "${ML_ADMIN_USERNAME}":"${ML_ADMIN_PASSWORD}" -m 30 -s --retry 5 -X GET http://localhost:8001/admin/v1/server-config | grep "host-id" | sed 's%^.*<host-id.*>\(.*\)</host-id>.*$%\1%')
```

Comment:
I think it's better to just have have a function that gets the host id. As the logic here is repetitive.